Tests/js/models/assets/ez-trashitemmodel-tests.js

```diff
@@ -204,8 +205,48 @@ YUI.add('ez-trashitemmodel-tests', function (Y) {
         },
     });
 
+    restoreTest = new Y.Test.Case({
+        name: "eZ TrashItem Model restore test",
+
+        setUp: function () {
+            this.model = new Y.eZ.TrashItem();
+            this.id = 42;
+
+            this.contentServiceMock = new Mock();
+            Mock.expect(this.contentServiceMock, {
+                method: "recover",
+                args: [this.id, Mock.Value.Function],
+                run: function (id, callback) {
+                    callback();
+                },
+            });
+
+            this.apiMock = new Mock();
+            Mock.expect(this.apiMock, {
+                method: "getContentService",
+                args: [],
+                returns: this.contentServiceMock
+            });
+
+            this.options = {api: this.apiMock};
+        },
+
+        tearDown: function () {
+            this.model.destroy();
+            delete this.model;
+        },
+
+        "Should call the api on restore": function () {
+            this.model.set('id', this.id);
+            this.model.restore(this.options, function () {});
```

Comment:
this test is incomplete. You should check that the model restore callback parameter is passed to the content service recover method (ie the contentService mock should be mocked here)